service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.connect.persistence.rpt

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright (c) 2007-2014, Kaazing Corporation. All rights reserved.
+#
+
+#
+# ============= client <-> gateway request 1
+#
+connect tcp://localhost:8110
+connected
+
+write "GET / HTTP/1.1\r\n"
+write "Host: localhost:8110\r\n"
+write "Cache-Control: max-age=0\r\n"
+write "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8\r\n"
+write "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.94 Safari/537.36\r\n"
+write "Accept-Encoding: gzip, deflate, sdch\r\n"
+write "Accept-Language: en-US,en;q=0.8\r\n"
+write "\r\n"
+
+read "HTTP/1.1 200 OK\r\n"
+read "Content-Type: text/html;charset=UTF-8\r\n"
+read /Date: .*/ "\r\n"
+read "Server: Kaazing Gateway\r\n"
+read "Transfer-Encoding: chunked\r\n"
+read "\r\n"
+read "14\r\n"
+read "<html>Helloo1</html>\r\n"
+read "0\r\n"
+read "\r\n"
+
```

Comment:
This barrier is implicitly there, writes after reads will always wait for the reads to complete